### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cs-renuka-andhale**
Descriptions:
PMDB Link: https://pmdb.fortinet.com/ProjectManagement/viewDocument.php?id=22907&doc_type=design

UTCs:
 Connector installation verified.
 Connector logo verified.
 Actions and Playbooks list verified.
 Playbooks tags verified.
 All Playbooks are in info mode verified.
 All Playbooks are in inactive mode verified.

### CODE REVIEW

**Comment by cs-suyog-jinturkar** [comment]
@cs-renuka-andhale, have you missed testing connector config, connector actions and sample playbooks? Or missed from mentioning them in your UTC?

**Comment by cs-renuka-andhale** [comment]
> @cs-renuka-andhale, have you missed testing connector config, connector actions and sample playbooks? Or missed from mentioning them in your UTC?

This is uncertified one as we do not have setup for this.

File: akamai/info.json [hunk 1]
```diff
@@ -0,0 +1,307 @@
+{
+  "name": "akamai",
+  "label": "Akamai",
```
Comment (cs-sneha-bharote) [inline akamai/info.json]: Update "Akamai" to "Akamai Prolexic"

File: akamai/info.json [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+  "description": "Akamai is an American content delivery network (CDN), cybersecurity, and cloud service company, providing web and Internet security services.",
+  "publisher": "Community",
+  "cs_approved": false,
+  "cs_compatible": true,
+  "version": "1.0.0",
+  "category": "Investigation",
+  "icon_small_name": "Akamai_small.png",
+  "icon_large_name": "Akamai_large.png",
```
Comment (cs-sneha-bharote) [inline akamai/info.json]: "icon_large_name": "Akamai_medium.png",

File: akamai/info.json [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+        "title": "Server URL",
+        "required": true,
+        "editable": true,
+        "visible": true,
+        "type": "text",
+        "name": "server_url",
+        "description": "URL of the Akamai server to which you will connect and perform the automated operations."
+      },
+      {
+        "title": "Client token",
```
Comment (cs-sneha-bharote) [inline akamai/info.json]: Client token/secret: make camel case

File: akamai/info.json [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+        {
+          "title": "Contract Name",
+          "required": true,
+          "editable": true,
+          "visible": true,
+          "type": "text",
+          "name": "contract",
+          "value": "",
+          "tooltip": "Contract name. e.g. coral",
+          "description": "The policy domain name of the data center or proxy that events belong to."
```
Comment (cs-sneha-bharote) [inline akamai/info.json]: "description": "The contract is the policy domain name of the data center or proxy that events belong to."

File: akamai/info.json [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+      "enabled": true,
+      "parameters": [
+        {
+          "title": "Contract Name",
+          "required": true,
+          "editable": true,
+          "visible": true,
+          "type": "text",
+          "name": "contract",
+          "value": "",
```
Comment (cs-sneha-bharote) [inline akamai/info.json]: remove empty value.

File: akamai/info.json [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+      "parameters": [
+        {
+          "title": "Attack ID",
+          "required": true,
+          "editable": true,
+          "visible": true,
+          "type": "integer",
+          "name": "attackId",
+          "value": "",
+          "tooltip": "Contract name. e.g. coral",
```
Comment (cs-sneha-bharote) [inline akamai/info.json]: update/remove tooltip.
where you get attack ID?

File: akamai/operations.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        headers = {"Accept": "application/json"}
+        try:
+            response = session_obj.request(method=method, url=url, params=params, data=data, json=json,
+                                           headers=headers,
+                                           verify=self.verify_ssl)
+
+            if response.ok:
+                result = response.json()
+                return result
+            elif messages_codes[response.status_code]:
```
Comment (cs-sneha-bharote) [inline akamai/operations.py]: elif messages_codes.get(response.status_code):

File: akamai/playbooks/playbooks.json [hunk 8]
```diff
@@ -0,0 +1,350 @@
+{
+  "type": "workflow_collections",
+  "data": [
+    {
+      "uuid": "18eae72e-136b-4d88-a054-5583230a2d23",
+      "@type": "WorkflowCollection",
+      "name": "Sample - Akamai - 1.0.0",
+      "description": "Akamai is an American content delivery network (CDN), cybersecurity, and cloud service company, providing web and Internet security services.",
+      "visible": true,
+      "image": null,
```
Comment (cs-sneha-bharote) [inline akamai/playbooks/playbooks.json]: "image": "Akamai_large.png",

100x100 logo image in playbook image.

File: akamai/playbooks/playbooks.json [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+            },
+            {
+              "uuid": "f74b9594-25b4-47cf-8c7a-7b506f0a455b",
+              "@type": "WorkflowStep",
+              "name": "List Critical Events",
+              "description": null,
+              "status": null,
+              "arguments": {
+                "name": "Akamai",
+                "config": "''",
```
Comment (cs-sneha-bharote) [inline akamai/playbooks/playbooks.json]: remove '' from config. 
Specify sample value of the contract.

File: akamai/info.json [hunk 10]
```diff
@@ -0,0 +1,307 @@
+{
+  "name": "akamai",
+  "label": "Akamai",
+  "description": "Akamai is an American content delivery network (CDN), cybersecurity, and cloud service company, providing web and Internet security services.",
+  "publisher": "Community",
+  "cs_approved": false,
+  "cs_compatible": true,
+  "version": "1.0.0",
+  "category": "Investigation",
```
Comment (cs-sneha-bharote) [inline akamai/info.json]: update category of the connector.

**Review by cs-sneha-bharote — APPROVED** [review]
LGTM.

**Pull request merged**